manager/controllers/app/fybrik_notebook_test.go

```diff
@@ -0,0 +1,204 @@
+			log.Println(fmt.Errorf("failed to upload file, %v", err))
+			t.Fatal(err)
+		}
+		if result != nil {
+			fmt.Printf("file uploaded to, %s\n", result.Location)
+		}
+	} else {
+		Expect(object).ToNot(BeNil())
+		log.Println("Object already exists in S3!")
+	}
+
+	err = apiv1alpha1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	k8sClient, err := client.New(ctrl.GetConfigOrDie(), client.Options{Scheme: scheme.Scheme})
+	Expect(err).To(BeNil())
+
+	// Create Kubernetes objects for test
+	// - namespace (in setup before)
+	// - asset (in setup before)
+	// - asset secret (in setup before)
+	// - arrow flight module (in setup before)
+	// - rego policy (in setup before)
+
+	// Push read module and application
+
+	// Module installed by setup script directly from remote arrow-flight-module repository
+	// Installing application
+	application := &apiv1alpha1.FybrikApplication{}
+	Expect(readObjectFromFile("../../testdata/notebook/fybrikapplication.yaml", application)).ToNot(HaveOccurred())
+	applicationKey := client.ObjectKeyFromObject(application)
+
+	// Create FybrikApplication and FybrikModule
+	Expect(k8sClient.Create(context.Background(), application)).Should(Succeed())
+
+	// Ensure getting cleaned up after tests finish
+	defer func() {
+		application := &apiv1alpha1.FybrikApplication{ObjectMeta: metav1.ObjectMeta{Namespace: applicationKey.Namespace, Name: applicationKey.Name}}
+		_ = k8sClient.Get(context.Background(), applicationKey, application)
+		_ = k8sClient.Delete(context.Background(), application)
+	}()
+
+	By("Expecting application to be created")
+	Eventually(func() error {
+		return k8sClient.Get(context.Background(), applicationKey, application)
+	}, timeout, interval).Should(Succeed())
+	By("Expecting plotter to be constructed")
+	Eventually(func() *apiv1alpha1.ResourceReference {
+		_ = k8sClient.Get(context.Background(), applicationKey, application)
+		return application.Status.Generated
+	}, timeout, interval).ShouldNot(BeNil())
+
+	// The plotter has to be created
+	plotter := &apiv1alpha1.Plotter{}
+	plotterObjectKey := client.ObjectKey{Namespace: application.Status.Generated.Namespace, Name: application.Status.Generated.Name}
+	By("Expecting plotter to be fetchable")
+	Eventually(func() error {
+		return k8sClient.Get(context.Background(), plotterObjectKey, plotter)
+	}, timeout, interval).Should(Succeed())
+
+	By("Expecting application to be ready")
+	Eventually(func() bool {
+		err := k8sClient.Get(context.Background(), applicationKey, application)
+		if err != nil {
+			return false
+		}
+		return application.Status.Ready
+	}, timeout, interval).Should(Equal(true))
+
+	// Forward port of arrow flight service to local port
+	svcName := strings.Replace(application.Status.AssetStates["fybrik-notebook-sample/data-csv"].Endpoint.Hostname, ".fybrik-blueprints.svc.cluster.local", "", 1)
+
+	command := exec.Command("kubectl", "-n", BlueprintNamespace, "port-forward", "svc/"+svcName, "8083:80")
```

Comment:
the port also should be taken from the endpoint?

Reply:
No, it is port-forward to the port above